packages/vscode/src/plugins/prisma-postgres-manager/PrismaPostgresRepository.ts

```diff
@@ -214,30 +215,62 @@ export class PrismaPostgresApiRepository implements PrismaPostgresRepository {
   }
 
   async getWorkspaces(): Promise<Workspace[]> {
+    if (this.workspacesCache.size > 0) return Array.from(this.workspacesCache.values())
+
     const credentials = await this.credentialsStore.getCredentials()
-    return credentials.map((cred) => ({
-      type: 'workspace',
-      id: cred.workspaceId,
-      name: `Personal Workspace ${cred.workspaceId}`, // TODO: pending /me endpoint api
-      token: cred.token,
-    }))
+    const results = await Promise.allSettled(
+      credentials.map(async (cred) => {
+        const client = await this.getClient(cred.workspaceId)
+        const response = await client.GET('/workspaces')
+        this.checkResponseOrThrow(cred.workspaceId, response)
+        const workspace: Workspace = {
+          type: 'workspace',
+          id: cred.workspaceId,
+          name: response.data.data[0].displayName,
+        }
+        this.workspacesCache.set(workspace.id, workspace)
+        return workspace
+      }),
+    )
+    return results
+      .flatMap((r) => (r.status === 'fulfilled' ? [r.value] : []))
+      .sort((a, b) => a.name.localeCompare(b.name))
   }
 
   async addWorkspace({ token, refreshToken }: { token: string; refreshToken: string }): Promise<void> {
-    const fakeId = token.substring(0, 8)
-    await this.credentialsStore.storeCredentials({
-      userId: 'prisma', // TODO: pending /me endpoint api
-      workspaceId: fakeId,
-      token,
-      refreshToken,
+    const client = createManagementAPIClient(token, () => {
```

Comment:
ℹ️ Cannot use the "usual" way to get a client here as we don't have the wokrspace id yet and haven't stored the credentials in the credentials store yet.

Am not super happy with this yet and want to look into refactoring the overall client handling in a dedicated PR. 🤔